### ISSUE TITLE
Error in building on Raspberry Pi 4

### ISSUE THREAD

**Issue by lucaturchet** [SkAT-VG/SDT#22]
Dear SDT team,
when building the sources on a Raspberry Pi 4 I get the following errors

`In file included from ../../src/Pd/bouncing~.c:1:
../../src/Pd/bouncing~.c: In function 'bouncing_new':
../../src/Pd/SDTCommonPd.h:25:3: error: implicit declaration of function 'error'; did you mean 'perror'? [-Werror=implicit-function-declaration]
   25 |   error("sdt.%s: unused argument in position %ld", s->s_name, uarg); \
      |   ^~~~~
../../src/Pd/bouncing~.c:51:3: note: in expansion of macro 'SDT_PD_ARG_PARSE'
   51 |   SDT_PD_ARG_PARSE(1, A_SYMBOL)
      |   ^~~~~~~~~~~~~~~~
../../src/Pd/SDTCommonPd.h:53:44: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
   53 | #define SDT_PD_REGISTER(T, F, S, I) x->key = GET_ARG(I, atom_getsymbol, 0, ->s_name); \
      |                                            ^
../../src/Pd/bouncing~.c:56:3: note: in expansion of macro 'SDT_PD_REGISTER'
   56 |   SDT_PD_REGISTER(Bouncing, bouncing, "bouncing process", 0)
      |   ^~~~~~~~~~~~~~~
In file included from ../../src/Pd/breaking~.c:1:
../../src/Pd/breaking~.c: In function 'breaking_new':
../../src/Pd/SDTCommonPd.h:25:3: error: implicit declaration of function 'error'; did you mean 'perror'? [-Werror=implicit-function-declaration]
   25 |   error("sdt.%s: unused argument in position %ld", s->s_name, uarg); \
      |   ^~~~~
../../src/Pd/breaking~.c:59:3: note: in expansion of macro 'SDT_PD_ARG_PARSE'
   59 |   SDT_PD_ARG_PARSE(1, A_SYMBOL)
      |   ^~~~~~~~~~~~~~~~
../../src/Pd/SDTCommonPd.h:53:44: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
   53 | #define SDT_PD_REGISTER(T, F, S, I) x->key = GET_ARG(I, atom_getsymbol, 0, ->s_name); \
      |                                            ^`

Such errors appear to me unrelated to the fact that the targeted platform is a Raspberry Pi, they seem issues of the syntax itself. Could you please provide some guidance?

**Comment by ChromaticIsobar** [comment]
Sorry for the delay.
I tried to replicate on my Raspberry Pi 3, without success.
I could replicate the issue on a Raspberry Pi 4 that I borrowed.

It seems that the issue is related to different versions of the Pd header.

The changes in branch [ChromaticIsobar:fix/issue22](https://github.com/ChromaticIsobar/SDT/tree/fix/issue22) fix the issue for my system.
I kindly ask you to verify if they solve the issue for your system.
